### PR TITLE
feat: Set up sass, webpack reload, server tests

### PR DESCRIPTION
**Pull request by DukeBaird**
- Update webpack with working sass setup
- Set up `npm run pack:watch` script for automatic re-bundling on save
- Sample layout for `jest` server tests
- Updated readme & dependencies

### CODE REVIEW

File: client/index.pug [hunk 1]
```diff
@@ -5,8 +5,5 @@ html
 		body
 			#root
 
-			each cssFile in htmlWebpackPlugin.files.css
-				style !{compilation.assets[cssFile.substr(htmlWebpackPlugin.files.puglicPath.length)].source()}
-
 			each jsFile in htmlWebpackPlugin.files.js
-				script !{compilation.assets[jsFile.substr(htmlWebpackPlugin.files.publicPath.length)].source()}
\ No newline at end of file
+				script !{compilation.assets[jsFile.substr(htmlWebpackPlugin.files.publicPath.length)].source()}
```
Comment (mcraig567) [inline client/index.pug:9]: What's different here? New line vs no new line?

File: client/src/index.jsx [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import '../style.sass';
```
Comment (mcraig567) [inline client/src/index.jsx:3]: What does ../style do vs ./style?

File: client/index.pug (hunk 1 above)
Comment (DukeBaird) [reply client/index.pug:9]: Yep, files should end in an empty newline

File: client/src/index.jsx (hunk 2 above)
Comment (DukeBaird) [reply client/src/index.jsx:3]: `./foo` means file `foo` in the same directory, where `../bar` means file `bar` in the parent directory, its just going back one level

**Review by mcraig567 — APPROVED** [review]

**Pull request merged**